ballerina/utils.bal

```diff
@@ -353,6 +353,45 @@ isolated function populateOptionalParameters(map<string> queryParamsMap, string?
     return queryParamsStr;
 }
 
+isolated function populateMultipartUploadHeaders(
+        map<string> requestHeaders,
+        MultipartUploadHeaders? multipartUploadHeaders) {
+    if multipartUploadHeaders is () {
+        return;
+    }
+    if multipartUploadHeaders?.cacheControl is string {
+        requestHeaders[IF_MODIFIED_SINCE] = <string>multipartUploadHeaders?.cacheControl;
+    }
```

Comment:
Can't we directly do the below using type narrowing?

```suggestion
    if multipartUploadHeaders?.cacheControl is string {
        requestHeaders[IF_MODIFIED_SINCE] = multipartUploadHeaders?.cacheControl;
    }
```

Reply:
It says `incompatible types: expected 'string', found 'string?'`

Will need to change the `MultipartUploadHeaders` record fields from `string?` to `string`
but then it will not let headers be optional.

Reply:
I could reproduce the same. @MaryamZi not sure if this is a known limitation or (not allowed by design) in our type narrowing implementation?

Reply:
Type narrowing happens only for parameters (which are implicitly final) and local variables. We don't narrow here since we can't guarantee the field to be of the specific type within the block also (since it may be mutated elsewhere).

You can assign `multipartUploadHeaders?.cacheControl` to a variable and use that in the check and assignment.

Reply:
As per maryam this is the expected behaviour as values are mutable. Therefore +1 to keep as is

Reply:
I would change to use the variable, since with the cast we still have the problem we are trying to avoid.